### PR TITLE
recipes-support: phytec-board-config: Restart awsclient after onboarding

### PR DESCRIPTION
**Pull request by ctenruh-phytec**
Restart awsclient three times after onboarding, so the necessary config
files are initialized.

### CODE REVIEW

**Comment by ctenruh-phytec** [comment]
do not merge yet, needs some changes. thanks.

**Comment by mschwan-phytec** [comment]
Mark this PR as "ready for review" when appropriate.

**Comment by ctenruh-phytec** [comment]
running the commands in the backround would not be ideal since the user would be able to close the phytec-board-config UI which will end the script and the awsclient would not be restarted properly.

File: recipes-support/provisioning/files/phytec-board-config.sh
Comment (mschwan-phytec) [inline recipes-support/provisioning/files/phytec-board-config.sh:45]: Is it necessary to change the height with this PR? I guess there are not  any UI changes that would require this.

File: recipes-support/provisioning/files/phytec-board-config.sh
Comment (mschwan-phytec) [inline recipes-support/provisioning/files/phytec-board-config.sh]: The while loop is executed during the non-interactive part, but you add user interaction to it. I'd suggest restarting the awsclient without asking in any case, if it is necessary. This also means that the function `do_restart_awsclient` can be dissolved into the code part here because it is not used elsewhere.

File: recipes-support/provisioning/files/phytec-board-config.sh
Comment (mschwan-phytec) [inline recipes-support/provisioning/files/phytec-board-config.sh]: See the other comment first, this change can probably be reverted.

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]

File: recipes-support/provisioning/files/phytec-board-config.sh
Comment (ctenruh-phytec) [reply recipes-support/provisioning/files/phytec-board-config.sh:45]: i did some minor UI changes in the text.

File: recipes-support/provisioning/files/phytec-board-config.sh
Comment (ctenruh-phytec) [reply recipes-support/provisioning/files/phytec-board-config.sh]: this while loop is needed since it is possible to execute an onboarding via commandline. Without this prompt the restart process would start immediately after generating the token and UID, so the user won't have the time to add the device on the platform.

Comment (mschwan-phytec) [reply recipes-support/provisioning/files/phytec-board-config.sh]: I understand. But still, we cannot ask for user input in a non-interactive program. This makes scripting `phytec-board-config` with the non-interactive options impossible. I would suggest adding an extra option that restarts the `awsclient` separately in this case. For the interactive UI we can add a part that asks for user input though.

File: recipes-support/provisioning/files/phytec-board-config.sh
Comment (mschwan-phytec) [inline recipes-support/provisioning/files/phytec-board-config.sh]: I think you can refactor this a little bit. The if statement does not seem to be required, as you change `$awsrestart` only in one place. Simply execute `do_restart_awsclient` directly in the interactive execution branch of the onboarding function's if-else statement. Do not forget to remove the variable `$awsrestart` after this.

File: recipes-support/provisioning/files/phytec-board-config.sh
Comment (mschwan-phytec) [inline recipes-support/provisioning/files/phytec-board-config.sh]: Use the imperative form here, capitalize the beginning of sentences and add punctuation: "Press 'Ok'. The awsclient will be restarted three times."

File: recipes-support/provisioning/files/phytec-board-config.sh
Comment (mschwan-phytec) [inline recipes-support/provisioning/files/phytec-board-config.sh]: Minor rewording: "Restarting awsclient, this may take a while..."

File: recipes-support/provisioning/files/phytec-board-config.sh
Comment (mschwan-phytec) [inline recipes-support/provisioning/files/phytec-board-config.sh]: Rephrasing: "After a successful onboarding the awsclient needs to be rerun with 'systemctl restart awsclient' three times."

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]

File: recipes-support/provisioning/files/phytec-board-config.sh
Comment (mschwan-phytec) [reply recipes-support/provisioning/files/phytec-board-config.sh]: Ok, I just remembered our discussion about this.

The problem here still is though, that the restart occurs immediately after a token is being requested, which is not ideal. The awsclient should be restarted after the phytec-board-config tool finishes to give the user enough time for onboarding their device.

Comment (ctenruh-phytec) [reply recipes-support/provisioning/files/phytec-board-config.sh]: in the interactive mode the restart only occurs when the user presses "ok" after the token was generated it also says so in the tool. The if statement is needed, because otherwise when the user onboards via commandline the awsclient would restart immediately after the token is generated and as you said that would not make sense. I wrote a prompt so that would not happen but we decided that its not good to have the user interact when using commandline option. So now the awsclient will not be restarted after onboarding via commandline, merely a note will be displayed that the awsclient needs to be restarted.

Comment (mschwan-phytec) [reply recipes-support/provisioning/files/phytec-board-config.sh]: Ok, but you can still remove that if statement and move the restarting of the service to the interactive part. And there is also no need to create the function `do_restart_awsclient` as it is only used once.

File: recipes-support/provisioning/files/phytec-board-config.sh
Comment (mschwan-phytec) [inline recipes-support/provisioning/files/phytec-board-config.sh]: You can remove this variable now.

**Review by mschwan-phytec — CHANGES_REQUESTED** [review]
Just one small thing before we can merge this...

**Review by mschwan-phytec — APPROVED** [review]

**Pull request merged**
